aptible/resource_app.go

```diff
@@ -186,6 +364,37 @@ func resourceAppRead(d *schema.ResourceData, meta interface{}) error {
 		log.Printf("ZDD flags: %t, %t", s.ForceZeroDowntime, s.NaiveHealthCheck)
 		service["force_zero_downtime"] = s.ForceZeroDowntime
 		service["simple_health_check"] = s.NaiveHealthCheck
+		// Find service_sizing_policy if any
+		var policy *aptibleapi.ServiceSizingPolicy
+		policy, err = findServiceSizingPolicyForService(s.Id, ctx, meta)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+		if policy != nil {
+			serviceSizingPolicy := make(map[string]interface{})
+			serviceSizingPolicy["autoscaling_type"] = policy.Autoscaling
+			serviceSizingPolicy["metric_lookback_seconds"] = policy.MetricLookbackSeconds
+			serviceSizingPolicy["percentile"] = formatFloat32ToFloat64(policy.Percentile)
+			serviceSizingPolicy["post_scale_up_cooldown_seconds"] = policy.PostScaleUpCooldownSeconds
+			serviceSizingPolicy["post_scale_down_cooldown_seconds"] = policy.PostScaleDownCooldownSeconds
+			serviceSizingPolicy["post_release_cooldown_seconds"] = policy.PostReleaseCooldownSeconds
+			serviceSizingPolicy["mem_cpu_ratio_r_threshold"] = formatFloat32ToFloat64(policy.MemCpuRatioRThreshold)
+			serviceSizingPolicy["mem_cpu_ratio_c_threshold"] = formatFloat32ToFloat64(policy.MemCpuRatioCThreshold)
+			serviceSizingPolicy["mem_scale_up_threshold"] = formatFloat32ToFloat64(policy.MemScaleUpThreshold)
+			serviceSizingPolicy["mem_scale_down_threshold"] = formatFloat32ToFloat64(policy.MemScaleDownThreshold)
+			serviceSizingPolicy["minimum_memory"] = policy.MinimumMemory
```

Comment:
Why do some of these use `GetX` and some not?

Reply:
The ones using `Get....` are NullableInts or NullableFloats, while the rest are direct Ints or Floats. Depends on whether they have default values and are NOT NIL on the ruby side. If I didn't use `GetX`, I'd get a pointer back that I would have to check if it was nil, and then dereference it - this is a shortcut